x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -208,15 +208,14 @@ export class TaskRunner<
 
   private async updateRuleSavedObjectPostRun(
     ruleId: string,
-    namespace: string | undefined,
     attributes: {
       executionStatus?: RawRuleExecutionStatus;
       monitoring?: RawRuleMonitoring;
       nextRun?: string | null;
       lastRun?: RawRuleLastRun | null;
     }
   ) {
-    const client = this.internalSavedObjectsRepository;
+    const client = this.context.elasticsearch.client.asInternalUser;
```

Comment:
Question: Does this require `asInternalUser`?

Reply:
We're updating the rule SO on behalf of the task runner, not on behalf of a user, so I think this usage is correct? WDYT?

Reply:
For these updates, feel like `asInternalUser` is appropriate, since it's the framework doing it.

Reply:
> We're updating the rule SO on behalf of the task runner, not on behalf of a user, so I think this usage is correct? WDYT?

> For these updates, feel like `asInternalUser` is appropriate, since it's the framework doing it.

@ymao1 @pmuellr ++ thanks! That makes sense.